zipline/lib/_windowtemplate.pxi

```diff
@@ -82,54 +85,61 @@ cdef class AdjustedArrayWindow:
         return self
 
     def __next__(self):
+        try:
+            self._tick_forward(1)
+        except Exhausted:
+            raise StopIteration()
+
+        self._update_output()
+        return self.output
+
+    def seek(self, Py_ssize_t target_anchor):
+        cdef:
+            Py_ssize_t anchor = self.anchor
+
+        if target_anchor < anchor:
+            raise Exception('Can not access data after window has passed.')
+
+        if target_anchor == anchor:
+            return self.output
+
+        self._tick_forward(target_anchor - anchor)
```

Comment:
Should `Exhausted` be handled here? Or is the intent to have that raise for the consumer of `seek` to handle?